core/src/main/scala/org/http4s/headers/Range.scala

```diff
@@ -49,7 +50,47 @@ object Range extends HeaderKey.Internal[Range] with HeaderKey.Singleton {
   }
 
   override def parse(s: String): ParseResult[Range] =
-    HttpHeaderParser.RANGE(s)
+    parser.parseAll(s).left.map { e =>
+      ParseFailure("Invalid Range header", e.toString)
+    }
+
+  val parser: P[Range] = {
+    // https://tools.ietf.org/html/rfc7233#section-3.1
+
+    val nonNegativeLong = Numbers.digits1
+      .map { ds =>
+        val l = BigInt(ds)
+        if (l < Long.MaxValue) l.toLong else Long.MaxValue
+      }
```

Comment:
How about a `mapFilter`?  (I'm doing this in GitHub. Hopefully it's right...)

```suggestion
    val nonNegativeLong = Numbers.digits1
      .mapFilter { ds =>
        try Some(ds.toLong)
        case { _: NumberFormatException => None }
      }
```

Reply:
Thanks, that looks better. I did not know mapFilter existed. I'll update both parsers.

Reply:
It's new in 0.2.0, and is based on the selective primitives, which makes it more efficient than a monadic parser.